daemon/service-db.cc

```diff
@@ -56,149 +74,277 @@ MLServiceDB::~MLServiceDB ()
 void
 MLServiceDB::connectDB ()
 {
-  char *err = nullptr;
-  leveldb_options_t *db_options;
+  int rc;
+  char *sql;
+  char *errmsg = nullptr;
 
-  if (db_obj)
+  if (_db != nullptr)
     return;
 
-  db_options = leveldb_options_create ();
-  leveldb_options_set_create_if_missing (db_options, 1);
+  rc = sqlite3_open (_path.c_str (), &_db);
+  if (rc != SQLITE_OK) {
+    g_warning ("Failed to open database: %s (%d)", sqlite3_errmsg (_db), rc);
+    goto error;
+  }
+
+  /* Create table and handle database version. */
+  if (!_initialized) {
+    /**
+     * @todo handle database version
+     * - check old level db and insert pipeline description into sqlite.
+     * - check database info (TBL_DB_INFO), fetch data and update table schema when version is updated, ...
+     * - need transaction
```

Comment:
Right we need to use `Transaction` as below.
```cpp
sqlite3_exec(_db, "BEGIN TRANSACTION;", NULL, NULL, NULL);
...
sqlite3_exec(_db, "END TRANSACTION;", NULL, NULL, NULL);
```

Reply:
updated